backend/src/Controller/Admin/StoreOwnerSubscription/AdminStoreSubscriptionController.php

```diff
@@ -99,4 +113,183 @@ public function deleteAllStoreFutureSubscriptionsByStoreOwnerId(int $storeOwnerI
 
         return $this->response($result, self::DELETE);
     }
+
+    /**
+     * admin: create subscription
+     * @Route("createsubscription", name="createSubscriptionByAdmin", methods={"POST"})
+     * @IsGranted("ROLE_ADMIN")
+     * @param Request $request
+     * @return JsonResponse
+     *
+     * @OA\Tag(name="Subscription")
+     *
+     * @OA\Parameter(
+     *      name="token",
+     *      in="header",
+     *      description="token to be passed as a header",
+     *      required=true
+     * )
+     *
+     * @OA\RequestBody(
+     *      description="create subscription",
+     *      @OA\JsonContent(
+     *          @OA\Property(type="string", property="status_code"),
+     *          @OA\Property(type="string", property="msg"),
+     *          @OA\Property(type="object", property="Data",
+     *              @OA\Property(type="integer", property="storeProfileId"),
+     *              @OA\Property(type="integer", property="packageId"),
+     *              @OA\Property(type="string", property="note"),
+     *      )
+     *    )
+     *  )
+     *
+     *
+     * @OA\Response(
+     *      response=201,
+     *      description="Returns subscription",
+     *      @OA\JsonContent(
+     *          @OA\Property(type="string", property="status_code"),
+     *          @OA\Property(type="string", property="msg"),
+     *          @OA\Property(type="object", property="Data",
+     *             @OA\Property(type="integer", property="id")
+     *      )
+     *    )
+     *  )
+     * 
+     * or
+     *
+     * @OA\Response(
+     *      response="default",
+     *      description="Return error.",
+     *      @OA\JsonContent(
+     *          @OA\Property(type="string", property="status_code", description="9351 or 9157"),
+     *          @OA\Property(type="string", property="msg", description="package not exist or store owner profile not exist!"),
+     *        )
+     *     )
+     * 
+     * @Security(name="Bearer")
+     */
+    public function createSubscription(Request $request): JsonResponse
```

Comment:
We may use another function name in order to differentiate it form createSubscription function that used by store owner.

Reply:
This function will only be called by (AdminStoreSubscriptionService)

Reply:
How a controller function is being called by a service one? :)